Soma-Technologies-Inc/visitAfrica-backend
# #175494062 User registration and email verification

#### What does this PR do?
- Reconfiguring sequelize
- Creating new user
- on successful registration of a user send him/her a verification email
#### Description of Task to be completed?
i have the these apis working
`GET /api/auth/signup`
#### How should this be manually tested?
- open your fancy terminal , in case you want to know what is a terminal visit this link [Terminal](https://en.wikipedia.org/wiki/Computer_terminal)
- git clone this repo by running `git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend.git`
- navigate to this branch by running `git checkout  ft-signupandemailverif-f-175494062`
- run `git pull ` to make sure that you have updated code code from this branch
- make a .env file and add JWTKEY , SENDGRID API KEY and POSTGRES DB Url
- run `npm run migrate` to create database migrations
- run `npm run startDev`  to start the server
- open your postman or an other software to test the api endpoints  in case you don't have postman installed download it on this link  [Postman](https://www.postman.com/)
- then make a request using postman to `http://localhost:3000/api/auth/signup`
- use a form like this `{
	"fullname": "elwin james",
	"email": "realemailaddress@gmail.com",
	"password": "AveryStrongPasswordLikeElysee!1!00",
	"birthdate": "1990-01-01",
	"gender":"Male",
	"tel": "0700000000",
	"country": "zimbabwe",
	"city": "Kigali"
}`

#### Any background context you want to provide?
#### Screenshots (if appropriate)
